operators/pkg/instance-controller/cloudinit_test.go

```diff
@@ -0,0 +1,298 @@
+		}
+
+		ctx, _ = clctx.InstanceInto(ctx, &instance)
+		ctx, _ = clctx.TemplateInto(ctx, &template)
+		ctx, _ = clctx.TenantInto(ctx, &tenant)
+		ctx, _ = clctx.EnvironmentInto(ctx, &environment)
+	})
+
+	Describe("The EnforceCloudInitSecret function", func() {
+		var expected []byte
+
+		Extractor := func(content map[string][]byte) string {
+			return string(content[instance_controller.UserDataKey])
+		}
+
+		BeforeEach(func() {
+			clientBuilder = *clientBuilder.WithObjects(ForgeWebDavSecret(
+				instance_controller.WebdavSecretUsernameKey, instance_controller.WebdavSecretPasswordKey))
+
+			expected, err = forge.CloudInitUserData(nextcloudBaseURL, webdavUsername, webdavPassword, tenant.Spec.PublicKeys)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		JustBeforeEach(func() {
+			err = reconciler.EnforceCloudInitSecret(ctx)
+		})
+
+		When("the secret does not yet exist", func() {
+			It("Should succeed", func() { Expect(err).ToNot(HaveOccurred()) })
+
+			It("Should be present and have the common attributes", func() {
+				Expect(reconciler.Get(ctx, objectName, &secret)).To(Succeed())
+				Expect(secret.GetLabels()).To(Equal(forge.InstanceObjectLabels(nil, &instance)))
+				Expect(secret.GetOwnerReferences()).To(ContainElement(ownerRef))
+			})
+
+			It("Should be present and have the expected content", func() {
+				Expect(reconciler.Get(ctx, objectName, &secret)).To(Succeed())
+				Expect(secret.Data).To(WithTransform(Extractor, Equal(string(expected))))
+				Expect(secret.Type).To(Equal(corev1.SecretTypeOpaque))
+			})
+		})
+
+		When("the secret already exists", func() {
+			BeforeEach(func() {
+				scrt := corev1.Secret{ObjectMeta: forge.NamespacedNameToObjectMeta(objectName)}
+				clientBuilder.WithObjects(&scrt)
+			})
+
+			It("Should succeed", func() { Expect(err).ToNot(HaveOccurred()) })
+
+			It("Should still be present and have the common attributes", func() {
+				Expect(reconciler.Get(ctx, objectName, &secret)).To(Succeed())
+				Expect(secret.GetLabels()).To(Equal(forge.InstanceObjectLabels(nil, &instance)))
+				Expect(secret.GetOwnerReferences()).To(ContainElement(ownerRef))
+			})
+
+			It("Should still be present and have unmodified content", func() {
+				Expect(reconciler.Get(ctx, objectName, &secret)).To(Succeed())
+				Expect(secret.Data).To(WithTransform(Extractor, Equal(string(expected))))
+				Expect(secret.Type).To(Equal(corev1.SecretTypeOpaque))
+			})
+		})
+
+	})
+
+	Describe("The GetWebDavCredentials function", func() {
+		var username, password string
+
+		JustBeforeEach(func() {
+			username, password, err = reconciler.GetWebDavCredentials(ctx)
+		})
+
+		Context("The webdav credentials secret does not exist", func() {
+			var notFoundError error
+
+			BeforeEach(func() {
+				notFoundError = kerrors.NewNotFound(corev1.Resource(string(corev1.ResourceSecrets)), webdavSecretName.Name)
+			})
+
+			It("Should return a not found error", func() { Expect(err).To(MatchError(notFoundError)) })
```

Comment:
Isn't the new matcher applicable here?

Reply:
Definitely. That was written before the matcher existed